pkg/scale/encode.go

```diff
@@ -230,23 +303,38 @@ func (es *encodeState) encodeArray(in interface{}) (err error) {
 func (es *encodeState) encodeBigInt(i *big.Int) (err error) {
 	switch {
 	case i == nil:
-		err = fmt.Errorf("nil *big.Int")
+		err = ErrEncodeBigInt
```

Comment:
Change `ErrEncodeBigInt` to `ErrBigIntIsNil` (or similar) and then wrap with

```suggestion
		err = fmt.Errorf("%w", ErrBigIntIsNil)
```